crml/rewards/src/lib.rs

```diff
@@ -0,0 +1,727 @@
+	where
+		Balance = BalanceOf<T>,
+	{
+		/// A reward payout happened (payout, remainder)
+		RewardPayout(Balance, Balance),
+	}
+);
+
+decl_storage! {
+	trait Store for Module<T: Trait> as Rewards {
+		/// Inflation rate % to apply on reward payouts, it may be negative
+		pub InflationRate get(fn inflation_rate): FixedI128 = FixedI128::saturating_from_integer(1);
+		/// Development fund % take for reward payouts, parts-per-billion
+		pub DevelopmentFundTake get(fn development_fund_take) config(): Perbill;
+		/// Accumulated transaction fees for reward payout
+		pub TransactionFeePot get(fn transaction_fee_pot): BalanceOf<T>;
+		/// Historic accumulated transaction fees on reward payout
+		pub TransactionFeePotHistory get(fn transaction_fee_pot_history): VecDeque<BalanceOf<T>>;
+	}
+}
+
+decl_module! {
+	pub struct Module<T: Trait> for enum Call where origin: T::Origin {
+
+		fn deposit_event() = default;
+
+		/// Set the per payout inflation rate (`numerator` / `denominator`) (it may be negative)
+		#[weight = (10_000, DispatchClass::Operational)]
+		pub fn set_inflation_rate(origin, numerator: i64, denominator: i64) {
+			ensure_root(origin)?;
+			if denominator.is_zero() {
+				return Err("denominator cannot be zero".into());
+			}
+			InflationRate::put(FixedI128::saturating_from_rational(numerator, denominator));
+		}
+
+		/// Set the development fund take %, capped at 100%.
+		#[weight = (10_000, DispatchClass::Operational)]
+		pub fn set_development_fund_take(origin, new_take_percent: u32) {
+			ensure_root(origin)?;
+			DevelopmentFundTake::put(
+				Perbill::from_percent(new_take_percent) // `from_percent` will saturate at `100`
+			);
+		}
+	}
+}
+
+impl<T: Trait> StakerRewardPayment for Module<T>
+where
+	BalanceOf<T>: FixedPointOperand,
+{
+	type AccountId = T::AccountId;
+	type Balance = BalanceOf<T>;
+	/// Perform a reward payout given a mapping of validators and their nominators stake.
+	/// Accounts IDs are the ones which should receive payment.
+	fn make_reward_payout(
+		validator_commission_stake_map: &[(Self::AccountId, Perbill, Exposure<Self::AccountId, Self::Balance>)],
+	) {
+		// Calculate the accumulated tx fee reward split
+		let fee_payout = TransactionFeePot::<T>::take();
+		// track historic era fee amounts
+		Self::note_fee_payout(fee_payout);
+
+		if fee_payout.is_zero() || validator_commission_stake_map.len().is_zero() {
+			return;
+		}
+
+		let mut total_payout = Self::inflation_rate().saturating_mul_acc_int(fee_payout);
+
+		// Deduct development fund take %
+		let development_fund_payout = Self::development_fund_take() * total_payout;
+		let _ = T::CurrencyToReward::deposit_into_existing(
+			&T::TreasuryModuleId::get().into_account(),
+			development_fund_payout,
+		);
+		total_payout = total_payout.saturating_sub(development_fund_payout);
+
+		// Payout reward to validators and their nominators
+		let total_payout_share = total_payout / BalanceOf::<T>::from(validator_commission_stake_map.len() as u32);
```

Comment:
This https://github.com/cennznet/cennznet/pull/304/files#diff-8b6330393ec7c1d63275edb851ba0d647bc896357706fa9377fe211ae033f0abR120-R139 
could be an individual function for calculating the payout share to each validator and its nominators.